### PR TITLE
Fixing the generated Canonical Link

### PR DESCRIPTION
**Pull request by kile-lindgren**
In Kentico 11, I was experiencing an issue where the canonical urls (both rel="amphtml" and rel="canonical" only contained the root domain, and was incorrectly appending all the url friendly extensions to the URL.

I adjusted the code to return the CurrentAliasPath when DocumentContext.CurrentPageInfo.DocumentUrlPath is null or empty, instead of just null.

### CODE REVIEW

File: Kentico.AcceleratedMobilePages/AmpFilter.cs [hunk 1]
```diff
@@ -360,5 +360,22 @@ private void RemoveAttribute(HtmlDocument doc, string xPath, string attrName)
                 }
             }
         }
+
+        /// <summary>
+        /// Splits the Friendly URL Extension, and returns the first element
+        /// </summary>
+        private string GetFriendlyExtension()
```
Comment (OndrejVrsan) [inline Kentico.AcceleratedMobilePages/AmpFilter.cs]: This method can be "inlined" directly to the setting's property. 
Actually all usages of 
https://github.com/Kentico/kentico-amp/blob/02b78c239bde5f4ffbc86e31030c18ddb65035a8/Kentico.AcceleratedMobilePages/Settings.cs#L12
expect that this setting returns only one extension (which is also promised by name of the property)

Instead this method, you can just change getter of this property to
`SettingsKeyInfoProvider.GetValue(SiteContext.CurrentSiteName + ".CMSFriendlyURLExtension").Split(';')[0];`
and it'll do precisely the same job as this method.

To explain why it is not needed this amount of code, `"".Split(';')` always returns at least the original string, in this case empty string. Therefore it is not needed to check whether `extensions.Length > 0` or the length of extension array after the split is more than 0.

File: Kentico.AcceleratedMobilePages/AmpFilter.cs [hunk 2]
```diff
@@ -69,8 +69,8 @@ private string AppendAmpHtmlLink(string finalHtml)
         {
             string ampLink = (CMSHttpContext.Current.Request.IsSecureConnection ? Constants.P_HTTPS : Constants.P_HTTP) +
                              Settings.AmpFilterDomainAlias +
-                             (DocumentContext.CurrentPageInfo.DocumentUrlPath ?? DocumentContext.CurrentAliasPath) +
-                             Settings.CmsFriendlyUrlExtension;
+                             (!String.IsNullOrEmpty(DocumentContext.CurrentPageInfo.DocumentUrlPath) ? DocumentContext.CurrentPageInfo.DocumentUrlPath : DocumentContext.CurrentAliasPath) +
```
Comment (OndrejVrsan) [inline Kentico.AcceleratedMobilePages/AmpFilter.cs]: This is getting longer and it is used on two places... Could you please refactor the
`(!String.IsNullOrEmpty(DocumentContext.CurrentPageInfo.DocumentUrlPath) ? DocumentContext.CurrentPageInfo.DocumentUrlPath : DocumentContext.CurrentAliasPath)`
to a method?

File: README.md [hunk 3]
```diff
@@ -136,3 +136,5 @@ Tested with Kentico 11.0 (net46).
 ## [Questions & Support](https://github.com/Kentico/Home/blob/master/README.md)
 
 ## [License](https://github.com/Kentico/kentico-amp/blob/master/LICENSE.txt)
+
```
Comment (OndrejVrsan) [inline README.md]: Why there are these extra newlines?

**Comment by petrsvihlik** [comment]
@kile-lindgren @OndrejVrsan does any of you guys have time to implement the review notes so that we can close this PR?

**Pull request merged**